introduction.tex

```diff
@@ -1,31 +1,31 @@
 
-As in our previous study we selected the commonly used RMSD algorithm implemented in \package{MDAnalysis} as a typical test case.
+As in our previous study we selected the commonly used RMSD algorithm implemented in \package{MDAnalysis} as a typical use case.
 We used the single program multiple data (SPMD) paradigm to parallelize this algorithm on HPC resources.
-With SPMD, each process executes essentially the same code but on a different part of the data. 
-We used Python, a machine-independent, byte-code interpreted, object-oriented programming (OOP) language, which is well-established in HPC parallel environments \cite{GAiN}. 
-We show that there are two important performance parameters, $\tcomp/\tIO$ and $\tcomp/\tcomm$ which are the ratio of computational to I/O load, as measured by the time spent on the computation versus the time spent on reading data from the file system, and the ratio of computational to communication load respectively that determine whether we observe stragglers.
+With SPMD, each process executes essentially the same operations on different parts of the data. 
+We used Python, a machine-independent, byte-code interpreted, object-oriented programming (OOP) language, which is well-established in HPC parallel community~\cite{GAiN}. 
+We show that there are two important performance parameters, $\tcomp/\tIO$ and $\tcomp/\tcomm$ which are the ratio of computation to I/O load, as measured by the time spent on computation versus the time spent on reading data from the file system, and the ratio of computation to communication load respectively, that determine whether we observe stragglers.
 If the problem was compute-bound ($\tcomp/\tIO \gg 1$), the algorithm scaled very well without modifications, but otherwise strong scaling performance beyond a single node was poor.  
-For the algorithms with small $\tcomp/\tIO$, we needed to come up with strategies to improve scaling and overcome problems with stragglers.
+For algorithms with small $\tcomp/\tIO$, we needed to come up with strategies to improve scaling and overcome problems with stragglers.
 To a lesser degree, $\tcomp/\tcomm$ also determined scaling and performance, with $\tcomp/\tcomm < 1$ predicting poor performance.
 We show that communication and I/O are the two main scalability bottlenecks.
 
-Taking advantage of Global Arrays toolkit \cite{GA, GAiN}, we were able to reduce communication cost noticeably.
-However, our data showed that the file access I/O (POSIX \texttt{stat()}) became the cause for stragglers beyond a single node.
+Taking advantage of Global Arrays~\cite{GA, GAiN}, we were able to reduce communication cost noticeably.
+However, our data showed that the file access I/O (POSIX \texttt{open()}) became the cause for stragglers beyond a single node.
```

Comment:
I also think that `open()` should be here instead of `stat()` because this is the function that our code actually calls (actually, `fopen()` – maybe that is not the same as POSIX `open()`?)

@mkhoshle did you change it from `open()` to `stat()` based on the Darshan results?

Reply:
@orbeckst Yes, I did it based on Darshan results. Should not it be stat instead of open?

Reply:
See https://github.com/hpcanalytics/paper-hpc-py-parallel-mdanalysis/pull/16#issuecomment-456981604 – I would rewrite the sentence as

     However, our data showed that the initial opening of the trajectory file became the cause for stragglers beyond a single node.

Reply:
I am not sure how to address this. Feel free to keep it or I can make the required merge and bring it in sync with master.

Reply:
> I also think that `open()` should be here instead of `stat()` because this is the function that our code actually calls (actually, `fopen()` – maybe that is not the same as POSIX `open()`?)
> 
> @mkhoshle did you change it from `open()` to `stat()` based on the Darshan results?

Yes, I believe so.

Reply:
see https://github.com/hpcanalytics/paper-hpc-py-parallel-mdanalysis/pull/16#issuecomment-457028060